include/Player/WaterGun.hpp

```diff
@@ -0,0 +1,285 @@
+#ifndef WATERGUN_HPP
+#define WATERGUN_HPP
+
+#include <JSystem/JGeometry.hpp>
+#include <JSystem/JDrama/JDRGraphics.hpp>
+#include <Player/NozzleBase.hpp>
+#include <Player/NozzleTrigger.hpp>
+#include <JSystem/J3D/J3DGraphAnimator/J3DModel.hpp>
+#include <Player/Yoshi.hpp>
+#include <Player/MarioMain.hpp>
+
+struct NozzleJointData {
+    u8 flags; // TODO: This is likely an enum. 0x1 is used, 0x4 is disabled.
+    u8 jointIndex;
+};
+
+struct NozzleData {
+    /* 0x00 */ u32 _00;                        // Number of water streams emitted by this nozzle
+    /* 0x04 */ u32 _04;                        // Unused padding
+    /* 0x08 */ const char *mHelmetPath;        // Path to optional helmet model (used by underwater nozzle)
+    /* 0x0C */ const char *mPath;              // Path to base nozzle model directory
+    /* 0x10 */ const char *mBmdPath;           // Full path to nozzle's BMD model file
+    /* 0x14 */ u8 mNumEmitters;                // Number of water emitter points on the nozzle
+    /* 0x15 */ u8 _15;                         // Unknown/padding
+    /* 0x16 */ NozzleJointData mJoints[3];     // Array of joint data for the nozzle
+};
+
+// Data structure containing model paths and configuration for each FLUDD nozzle type
+class TNozzleBmdData {
+    public:
+    u8 getEmitterCount(int index) const {
+        return mNozzles[index].mNumEmitters;
+    }
+
+    u8 getFlags(int index, int jointIndex) const {
+        return mNozzles[index].mJoints[jointIndex].flags;
+    }
+
+    u8 getJointIndex(int index, int jointIndex) const {
+        return mNozzles[index].mJoints[jointIndex].jointIndex;
+    }
+    
+    NozzleData mNozzles[6];
+};
+// Define the global variable in .data section
+const static TNozzleBmdData nozzleBmdData = {
+    {
+        {
+            0,                                          // _00
+            0,                                          // _04
+            nullptr,                                    // mHelmetPath
+            "/mario/watergun2/normal_wg",              // mPath
+            "/mario/watergun2/normal_wg/normal_wg.bmd", // mBmdPath
+            1,                                          // mNumEmitters
+            2,                                          // _15
+            {
+                {1, 0},
+                {4, 0},
+                {4, 0}
+            }
```

Comment:
JFYI: before merging you'll have to format, and to make clang-format not uglify large initializers like this one what you need to do is to leave an oxford comma after the last element in each subobject